modin/experimental/core/execution/native/implementations/hdk_on_native/test/test_dataframe.py

```diff
@@ -1631,6 +1631,7 @@ def merge(df, df2, on_columns, **kwargs):
             on_columns="A",
             constructor_kwargs={"dtype": "category"},
             comparator=lambda df1, df2: df_equals(df1.astype(float), df2.astype(float)),
+            force_lazy=False,
```

Comment:
Operation `astype` is no longer lazy?

Reply:
Merge on float columns defaults to pandas. Previously all dictionary columns converted to string, that is not correct, but the test didn't fail. Currently, only the string dictionaries are converted to string (or not converted at all). All other dictionaries are converted to the value type.